src/internal/pjs/util.go

```diff
@@ -2,23 +2,25 @@ package pjs
 
 import (
 	"context"
+	"github.com/pachyderm/pachyderm/v2/src/internal/grpcutil"
+	"github.com/pachyderm/pachyderm/v2/src/storage"
 	"testing"
 
 	"google.golang.org/grpc"
 
 	"github.com/pachyderm/pachyderm/v2/src/auth"
 	"github.com/pachyderm/pachyderm/v2/src/pjs"
 
-	"github.com/pachyderm/pachyderm/v2/src/internal/grpcutil"
 	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
 )
 
 type ClientOptions func(env *Env)
 
-func NewTestClient(t testing.TB, db *pachsql.DB, opts ...ClientOptions) pjs.APIClient {
+func NewTestClient(t testing.TB, db *pachsql.DB, client storage.FilesetClient, opts ...ClientOptions) pjs.APIClient {
 	env := Env{
 		DB:               db,
 		GetPermissionser: &testPermitter{mode: permitterAllow},
+		GetStorageClient: func() storage.FilesetClient { return client },
```

Comment:
What is the purpose of doing so?

Reply:
good question. I tried a few different things like passing in a server directly. The problem is that we can't call some functions on the storage server as they take a GRPC server as a parameter (read fileset for example). 

That means we need a client. Now the problem is that you can't just provide a client directly because the client doesn't exist at the time that the servers are all being created. The client only exists once the servers are all instantiated and the gRPC servers start up.

Reply:
I modified this to take a ctx so each time a client is returned, it changes according to the ctx of the request.